ouroboros-consensus/ouroboros-consensus-test-infra/src/Test/ThreadNet/Network.hs

```diff
@@ -637,8 +640,38 @@ runThreadNetwork systemTime ThreadNetworkArgs
         ledger <- atomically $ ledgerState <$> getExtLedger
         -- Combine the node's seed with the current slot number, to make sure
         -- we generate different transactions in each slot.
-        let txs = runGen (nodeSeed `combineWith` unSlotNo curSlotNo) $
-                    testGenTxs numCoreNodes curSlotNo cfg txGenExtra ledger
+        let genTxs i =
+                runGen
+                  (nodeSeed `combineWith` unSlotNo curSlotNo `combineWith` i)
+                  (testGenTxs numCoreNodes curSlotNo cfg txGenExtra ledger)
+
+        -- These TxGen generators fail by invoking 'QC.discard'. We try many
+        -- times before actually giving up. When we give up, it fatal, not just
+        -- a discard.
+        let loop acc = do
+                let txs = genTxs acc
+                lr <- tryJust isDiscard $ MonadThrow.evaluate $ seqList txs
+                case lr of
+                  Right _ -> pure txs
+                  Left  _ ->
+                      if acc < accLimit then loop (acc + 1) else
+                      MonadThrow.throwM (TxGenFailure accLimit)
+              where
+                seqList :: forall a. [a] -> ()
+                seqList = \case
+                    []   -> ()
+                    x:xs -> x `seq` seqList xs
```

Comment:
There exists [`forceElemsToWHNF`](https://github.com/input-output-hk/cardano-prelude/blob/eee27bd6697d5ab1b943ee9c50ee638bde009c81/src/Cardano/Prelude/Strict.hs#L17) for this.

Reply:
Thanks!